### ISSUE TITLE
Create a reasonable package structure.

### ISSUE THREAD

**Issue by iocanel** [kubernetes-client/java#407]
At the moment the generated packages are pretty flat and information like apiGroups and apiVersion are part of the generated classes. This makes the generated code non-intuitive.

Also, when it comes to fluent builderrs, it affects generated method names, making things even worse.

It would be awesome if we could work around this issue either by some kind of configuration or using a custom language.

**Comment by brendandburns** [comment]
@iocanel I'm open to changing the directory structure. Can you describe the layout you had in mind? That pom.xml is a little hard to parse for me.

Also, do you have a sense for where this fits into the project? Should this be part of how we generate the code in the first place? (in which case it goes in the https://github.com/kubernetes-client/gen repo) or part of the client build process here (in which case this is the right repo)

Thanks!

**Comment by iocanel** [comment]
Here are some examples:

     io.kubernetes.client.models.v1.Pod
     io.kubernetes.client.models.apps.v1beta1.Deployment
     io.kubernetes.client.models.extensions.v1beta1.Scale

In other words when a type is defined as "apps.v1beta1.Deployment" it will trigger the generation of a class with fqcn "io.kubernetes.client.models.apps.v1beta1.Deployment".

**Comment by brendandburns** [comment]
That looks fine to me. Where do you want to add this processing?

We're about to cut a 4.0.0 so this is a good time to make such a breaking change...

**Comment by iocanel** [comment]
@brendanburns: I think that it needs to go to the gen project. The generated code will also include the annotation configuration for generating nested fluent builders, so  https://github.com/kubernetes-client/java/blob/master/kubernetes/src/main/java/io/kubernetes/client/fluent/Config.java will become obsolete. 

Its important to note that the generated models will also be immutable so no setters here.

Let me create a pull request on the gen project and we can further discuss this.
